recipes/xkbcommon/all/conanfile.py

```diff
@@ -70,80 +65,97 @@ def validate(self):
             raise ConanInvalidConfiguration("This library is only compatible with Linux or FreeBSD")
 
     def build_requirements(self):
-        self.build_requires("meson/0.63.1")
-        self.build_requires("bison/3.7.6")
-        if hasattr(self, "settings_build") and self.options.get_safe("wayland"):
-            self.build_requires("wayland/1.21.0")
+        self.tool_requires("meson/0.63.0")
+        self.tool_requires("bison/3.7.6")
+        self.tool_requires("pkgconf/1.7.4")
+        if self.options.get_safe("with_wayland"):
+            self.tool_requires("wayland/1.21.0")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def source(self):
-        files.get(self, **self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-        # Conan doesn't provide a `wayland-scanner.pc` file for the package in the _build_ context
-        files.replace_in_file(self, f"{self._source_subfolder}/meson.build",
-                              "wayland_scanner_dep = dependency('wayland-scanner', required: false, native: true)",
-                              "# wayland_scanner_dep = dependency('wayland-scanner', required: false, native: true)")
-
-        files.replace_in_file(self, f"{self._source_subfolder}/meson.build",
-                              "if not wayland_client_dep.found() or not wayland_protocols_dep.found() or not wayland_scanner_dep.found()",
-                              "if not wayland_client_dep.found() or not wayland_protocols_dep.found()")
-
-        files.replace_in_file(self, f"{self._source_subfolder}/meson.build",
-                              "wayland_scanner = find_program(wayland_scanner_dep.get_pkgconfig_variable('wayland_scanner'))",
-                              "wayland_scanner = find_program('wayland-scanner')")
-
-    def _configure_meson(self):
-        if self._meson:
-            return self._meson
-        defs={
-            "enable-wayland": self.options.get_safe("with_wayland", False),
-            "enable-docs": False,
-            "enable-x11": self.options.with_x11,
-            "libdir": f"{self.package_folder}/lib",
-            "default_library": ("shared" if self.options.shared else "static")}
-        if self._has_xkbregistry_option:
-            defs["enable-xkbregistry"] = self.options.xkbregistry
+    def generate(self):
+        tc = MesonToolchain(self)
+
+        tc.project_options["libexecdir"] = "bin"
+        tc.project_options["libdir"] = "lib"
+        tc.project_options["datadir"] = "res"
+        tc.project_options["mandir"] = "res/man"
```

Comment:
mandir could be remove, we usually don't use it. Is it really required, like, the library does not work without it?

Reply:
Aw, `enable-docs` is for generating Doxygen documentation. There's no option for disabling the man pages.

Reply:
just do `rmdir(self, os.path.join(self.package_folder, "share")` then

Reply:
Done.